hri_safety_sense/src/VscProcess.cpp

```diff
@@ -36,186 +35,215 @@
-		ROS_FATAL("Cannot open serial port! (%s, %i)",serialPort.c_str(),serialSpeed);
-	} else {
-		ROS_INFO("Connected to VSC on %s : %i",serialPort.c_str(),serialSpeed);
-	}
-
-	// Attempt to Set priority
-	bool  set_priority = false;
-	if(nh.getParam("set_priority", set_priority)) {
-		ROS_INFO("Set priority updated to:  %i",set_priority);
-	}
-
-	if(set_priority) {
-		if(setpriority(PRIO_PROCESS, 0, -19) == -1) {
-			ROS_ERROR("UNABLE TO SET PRIORITY OF PROCESS! (%i, %s)",errno,strerror(errno));
-		}
-	}
-
-	// Create Message Handlers
-	joystickHandler = new JoystickHandler(frameId);
-
-	// EStop callback
-	estopServ = rosNode.advertiseService("safety/service/send_emergency_stop", &VscProcess::EmergencyStop, this);
-
-	// KeyValue callbacks
-	keyValueServ = rosNode.advertiseService("safety/service/key_value", &VscProcess::KeyValue, this);
-	keyStringServ = rosNode.advertiseService("safety/service/key_string", &VscProcess::KeyString, this);
-
-	// Publish Emergency Stop Status
-	estopPub = rosNode.advertise<std_msgs::UInt32>("safety/emergency_stop", 10);
-
-	// Main Loop Timer Callback
-	mainLoopTimer = rosNode.createTimer(ros::Duration(1.0/VSC_INTERFACE_RATE), &VscProcess::processOneLoop, this);
-
-	// Init last time to now
-	lastDataRx = ros::Time::now();
-
-	// Clear all error counters
-	memset(&errorCounts, 0, sizeof(errorCounts));
+  std::string serialPort = "/dev/ttyACM0";
+  this->declare_parameter<std::string>("serial", serialPort);
+  if (this->get_parameter_or<std::string>("serial", serialPort, serialPort))
+    RCLCPP_INFO(this->get_logger(), "Serial Port updated to:  %s", serialPort.c_str());
+
+  int serialSpeed = 115200;
+  this->declare_parameter<int>("serial_speed", serialSpeed);
+  if (this->get_parameter_or<int>("serial_speed", serialSpeed, serialSpeed))
+    RCLCPP_INFO(this->get_logger(), "Serial Port Speed updated to:  %i", serialSpeed);
+
+  std::string frameId = "/srcs";
+  this->declare_parameter<std::string>("frame_id", frameId);
+  if (this->get_parameter_or<std::string>("frame_id", frameId, frameId))
+    RCLCPP_INFO(this->get_logger(), "Frame ID updated to:  %s", frameId.c_str());
+
+  /* Open VSC Interface */
+  vscInterface = vsc_initialize(serialPort.c_str(), serialSpeed);
+  if (vscInterface == NULL) {
+    RCLCPP_FATAL(this->get_logger(), "Cannot open serial port! (%s, %i)", serialPort.c_str(), serialSpeed);
+  } else {
+    RCLCPP_INFO(this->get_logger(), "Connected to VSC on %s : %i", serialPort.c_str(), serialSpeed);
+  }
+
+  // Attempt to Set priority
+  bool setPriority = false;
+  this->declare_parameter<bool>("set_priority", setPriority);
+  if (this->get_parameter_or<bool>("set_priority", setPriority, setPriority))
+    RCLCPP_INFO(this->get_logger(), "Set priority updated to:  %i", setPriority);
+
+  if(setPriority) {
+    if(setpriority(PRIO_PROCESS, 0, -19) == -1) {
+      RCLCPP_ERROR(this->get_logger(), "UNABLE TO SET PRIORITY OF PROCESS! (%i, %s)", errno, strerror(errno));
```

Comment:
I know this came from the old code, but I think if the user asked to set priority and it wasn't successful, this should abort the program (throw an exception).  Also, yelling at the user in all capitals isn't nice :).

Reply:
Done in 45091aa